go/vumitools/tasks.py

```diff
@@ -0,0 +1,53 @@
+import json
+from celery.task import task
+
+from go.base.command_utils import get_user_by_account_key
+from go.base.utils import vumi_api, vumi_api_for_user
+from go.config import get_conversation_definition
+from go.vumitools.metrics import get_conversation_metric_prefix
+
+
+@task(ignore_result=True)
+def send_recent_conversation_metrics():
+    conversation_details = get_and_reset_recent_conversations()
+
+    for conv_details in conversation_details:
+        details = json.loads(conv_details)
+        user_api = get_user_api(details["account_key"])
+        conv = get_conversation(details["account_key"], details["conv_key"])
+        publish_conversation_metrics(user_api, conv)
+
+
+def get_and_reset_recent_conversations():
+    redis = vumi_api().redis.sub_manager("conversation.metrics.middleware")
+
+    # makes use of redis atomic functions to ensure nothing is added to the set
+    # before it is deleted
+    redis.rename("recent_coversations", "old_recent_conversations")
+    conversation_details = redis.get("old_recent_conversations")
```

Comment:
I'd prefer to use `smembers` here, just so its explicit that we expect this to be a set.